src/client/Microsoft.Identity.Client/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -182,15 +184,16 @@ internal override JObject ToJObject()
             SetItemIfValueNotNull(json, StorageJsonKeys.Realm, TenantId);
             SetItemIfValueNotNull(json, StorageJsonKeys.Target, _scopes);
             SetItemIfValueNotNull(json, StorageJsonKeys.UserAssertionHash, UserAssertionHash);
+            SetItemIfValueNotNull(json, StorageJsonKeys.OboCacheKey, OboCacheKey);

```

Comment:
Do we need a new field in the token cache schema? I think we can reuse the UserAssertionHash. 

If the user gives us a key, we will always use it. Otherwise, we will use hash(incoming_assertion)